extensions/elytron-security-properties-file/runtime/src/main/java/io/quarkus/elytron/security/runtime/ElytronPropertiesFileRecorder.java

```diff
@@ -210,4 +211,24 @@ public void run() {
         SecurityRealm realm = new SimpleMapBackedSecurityRealm(NameRewriter.IDENTITY_REWRITER, providers);
         return new RuntimeValue<>(realm);
     }
+
+    private static String getAlgorithm(String algorithm) {
+        switch (algorithm) {
+            case "ALGORITHM_DIGEST_MD5":
+                return DigestPassword.ALGORITHM_DIGEST_MD5;
+            case "ALGORITHM_DIGEST_SHA":
+                return DigestPassword.ALGORITHM_DIGEST_SHA;
+            case "ALGORITHM_DIGEST_SHA_256":
+                return DigestPassword.ALGORITHM_DIGEST_SHA_256;
+            case "ALGORITHM_DIGEST_SHA_384":
+                return DigestPassword.ALGORITHM_DIGEST_SHA_384;
+            case "ALGORITHM_DIGEST_SHA_512":
+                return DigestPassword.ALGORITHM_DIGEST_SHA_512;
+            case "ALGORITHM_DIGEST_SHA_512_256":
+                return DigestPassword.ALGORITHM_DIGEST_SHA_512_256;
+            default:
```

Comment:
I'd consider throwing the exception instead, if the user expects some super secured algorithm then we can't really default to something else IMHO.

Reply:
Thanks for the feedback. Yes, That does make sense. Will update this.

Reply:
@sberyozkin I've updated the logic, if the algorithm doesn't match the predefined ones then we'll use it directly in PasswordFactory. This has 2 advantages
1) No need to build a custom exception, A predefined exception will be thrown by elytron if no such algorithm exists.
2) Users now can also use short form directly like "digest-md5".

Reply:
@lower-case should `algorithm` property be enum:
```
enum Algorithm {
    DIGEST_MD5("digest-md5")
    //etc
}
```
Otherwise `getAlgorithm` does not do anything actually, you can pass `algorithm` as is anyway